crates/bevy_transform/src/systems.rs

```diff
@@ -30,16 +46,19 @@ pub fn propagate_transforms(
         (Entity, &Children, Ref<Transform>, &mut GlobalTransform),
         Without<Parent>,
     >,
+    orphaned: RemovedComponents<Parent>,
     transform_query: Query<(Ref<Transform>, &mut GlobalTransform, Option<&Children>), With<Parent>>,
     parent_query: Query<(Entity, Ref<Parent>)>,
 ) {
+    let mut orphaned = orphaned.iter().collect::<Vec<_>>();
+    orphaned.sort_unstable();
     root_query.par_for_each_mut(
         // The differing depths and sizes of hierarchy trees causes the work for each root to be
         // different. A batch size of 1 ensures that each tree gets it's own task and multiple
         // large trees are not clumped together.
         1,
         |(entity, children, transform, mut global_transform)| {
-            let changed = transform.is_changed();
+            let changed = transform.is_changed() || orphaned.binary_search(&entity).is_ok();
```

Comment:
It may actually be faster to do a sparse set `contains` check here than do the binary search. It'd also avoid the allocation and sort.

Reply:
How would this look? `RemovedComponents` only exposes the `iter()` method. all other fields private. I guess the best way would be to add a method on `RemovedComponents` to allow sparse set contains checks.

Reply:
Looks like the entities in `World::removed_components` are stored in a `Vec`. It's unlikely that it's possible to outperform sorting + binary search. I expect the `Vec` to be always relatively small and not spill out of cache.

Reply:
This is non-actionable. If performance is a concern, could you point to ways of benchmarking testing for regression?

Since  the current transform benchmarks do not have parent/child removal/addition, I doubt it would help find any differences.

Reply:
> Another option is to store `orphaned` in a `Local` to avoid allocation at each frame. Of course, since the iterator returned by `RemovedComponents::iter` is `Cloned<slice::Iter<Entity>>`, I don't expect it to allocate when it's empty. And storing the `Vec` in a `Local` smells memory leak.

This would be a nice (but not necessary) optimization. It would just allow the system to reuse a single allocation, it shouldn't cause a memory leak or anything bad.

Reply:
My worry is that orphaning happens in bursts (think of unloading/loading scenes), so we end up with reserved memory for the worst case while the `Vec` is only fully used one frame every minutes/hours.

I think a cool thing bevy could provide is some sort of smart `Local<Vec<T>>` that clears intelligently. I think it would benefit the engine itself already.